vulnerabilities/importers/curl.py

```diff
@@ -0,0 +1,178 @@
+                package = data.get("database_specific").get("package")
+                logger.error(f"Invalid CVE ID: {cve_id} in package {package}")
+                continue
+            yield parse_advisory_data(data)
+
+
+def parse_advisory_data(raw_data) -> AdvisoryData:
+    """
+    Parse advisory data from raw JSON data and return an AdvisoryData object.
+
+    Args:
+        raw_data (dict): Raw JSON data containing advisory information.
+
+    Returns:
+        AdvisoryData: Parsed advisory data as an AdvisoryData object.
+
+    Example:
+        >>> raw_data = {
+        ...     "aliases": ["CVE-2024-2379"],
+        ...     "summary": "QUIC certificate check bypass with wolfSSL",
+        ...     "database_specific": {
+        ...         "package": "curl",
+        ...         "URL": "https://curl.se/docs/CVE-2024-2379.json",
+        ...         "www": "https://curl.se/docs/CVE-2024-2379.html",
+        ...         "issue": "https://hackerone.com/reports/2410774",
+        ...         "severity": "Low",
+        ...         "CWE": {
+        ...             "id": "CWE-297",
+        ...             "desc": "Improper Validation of Certificate with Host Mismatch"
+        ...        },
+        ...     },
+        ...     "published": "2024-03-27T08:00:00.00Z",
+        ...     "affected": [
+        ...         {
+        ...             "ranges": [
+        ...                 {
+        ...                     "type": "SEMVER",
+        ...                     "events": [
+        ...                         {"introduced": "8.6.0"},
+        ...                         {"fixed": "8.7.0"}
+        ...                     ]
+        ...                 }
+        ...             ],
+        ...             "versions": ["8.6.0"]
+        ...         }
+        ...     ]
+        ... }
+        >>> parse_advisory_data(raw_data)
+        AdvisoryData(aliases=['CVE-2024-2379'], summary='QUIC certificate check bypass with wolfSSL', affected_packages=[AffectedPackage(package=PackageURL(type='generic', namespace='curl.se', name='curl'), affected_version_range=GenericVersionRange(min_version=SemverVersion(version='8.6.0'), max_version=None), fixed_version=SemverVersion(version='8.7.0'))], references=[Reference(url='https://curl.se/docs/CVE-2024-2379.html', severities=[VulnerabilitySeverity(system='generic_textual', value='Low')])], date_published=datetime.datetime(2024, 3, 27, 8, 0, tzinfo=datetime.timezone.utc), weaknesses=["297"])
+
+    """
+    affected = get_item(raw_data, "affected")[0] if len(get_item(raw_data, "affected")) > 0 else []
+
+    ranges = get_item(affected, "ranges")[0] if len(get_item(affected, "ranges")) > 0 else []
+    events = get_item(ranges, "events")[1] if len(get_item(ranges, "events")) > 1 else {}
+    type_of_version = get_item(ranges, "type") if get_item(ranges, "type") else ""
+    # fixed_version = ""
+    if type_of_version=="SEMVER":
+        fixed_version = SemverVersion(events.get("fixed") or "")
```

Comment:
This seems like a cleaner way to do it.
```suggestion
    version_type = get_item(ranges, "type") if get_item(ranges, "type") else ""
    fixed_version = events.get("fixed")
    if version_type=="SEMVER" and fixed_version:
        fixed_version = SemverVersion(fixed_version)
```